lua/telescope/builtin.lua

```diff
@@ -833,4 +833,34 @@ builtin.marks = function(opts)
   }):find()
 end
 
+builtin.tagstack = function(opts)
+  opts = opts or {}
+  local tagstack = vim.fn.gettagstack()
+  if vim.tbl_isempty(tagstack.items) then
+    return
+  end
+
+  for i, value in pairs(tagstack.items) do
+    value.text = value.tagname
+    value.lnum = value.from[2]
+    value.filename = vim.fn.bufname(value.from[1])
+  end
+
+  -- reverse the list
+  tags = {}
+  for i=#tagstack.items, 1, -1 do
+    tags[#tags+1] = tagstack.items[i]
```

Comment:
you can use `table.insert(tags, tagstack.items[i])` for simpler lua.

Reply:
af3c348ce302fa71f9287aa3a2ed3668290c7274